### PR TITLE
reduce the number of translatable strings in the unit type editor

### LINKED ISSUE #8677: Review the translatable strings in the unit editor
This is logged as a reminder to do it before 1.19's eventual string freeze, probably 18-30 months from now; however I'm assuming there may be other changes before then.

Many of the strings in `editor_edit_unit.cfg` are only understandable in context, and so need PO hints.

Some of them might be mergeable with text from other areas, for example from the help brower's strings.

Some filenames have been marked translatable, for example
```
[widget]
	id = "tab_image"
	label = _ "/data/core/images/attacks/rectangular-shield.png"
[/widget]
```

### PR DESCRIPTION
**Pull request by babaissarkar**
Resolves #8677.

### CODE REVIEW

**Comment by babaissarkar** [comment]
Note that this leads to some inconsistent spacing and colons (:).

**Comment by CelticMinstrel** [comment]
Just checking, but you're aware that a textdomain holds until the next textdomain, right? Since there are some strings now that don't have a textdomain above them but are now loaded from a different textdomain than before.

(I was actually wondering if it would make sense to add a redundant textdomain declaration above them to make it clearer…? Or is that silly?)

**Comment by babaissarkar** [comment]
Yes, I know that the textdomain holds until the next one is declared. I'll
try to add extra declarations to make the intent clear.

**Comment by stevecotton** [comment]
We've got a long time before 1.20 releases. Unless this is specifically aiming for a backport to 1.18, I think it would be worth waiting to see if RTL support causes changes to the strings that end "colon space".

**Comment by babaissarkar** [comment]
> We've got a long time before 1.20 releases. Unless this is specifically aiming for a backport to 1.18, I think it would be worth waiting to see if RTL support causes changes to the strings that end "colon space".

So there's work ongoing on RTL support?

Some more explanation would be good. What would you suggest? postpone this until later?
I don't have any problems with backporting this to 1.18 if this helps the translators.

**Comment by stevecotton** [comment]
I mean, the unit editor isn't in 1.18, so these strings would only affect 1.18 if the unit editor itself was going to be backported - I remember discussing that, not sure if it's still happening or not.

While there's a lot of RTL UI improvements that could still be done, the Arabic translation is now over 80% complete, and [complete for every campaign except EI and TSG](https://gettext.wesnoth.org/?view=langs&version=branch&lang=ar). That raised the priority of #5600, and the help is now in a usable state, although the right-hand pane of the help shows the problems with having two labels next to each other, "heading: " "value", when the positioning of the labels expects an LTR language. I think the solution is going to be "heading: $value", so that the text direction works.

**Comment by babaissarkar** [comment]
> "heading: $value"
That's a good idea. Although Bengali is LTR, it still suffers from the same problem time to time, where having the values as variables would help.

I should probably keep RTL in mind for #8655, though the main issue lies in the help topic generators probably.

**Comment by Vultraz** [comment]
Why does this add spaces after colons in some cases?

**Comment by stevecotton** [comment]
@babaissarkar: if I understand correctly, for Bengali grammar you need something like "Becomes $value when promoted"; so you'll also want the coloration markup to be part of the translatable string, so that the unit name can have white text both before and after?

@Vultraz: because it's trying to reuse existing strings, and some of the existing ones have a space after the colon.

**Comment by babaissarkar** [comment]
@stevecotton 
> so you'll also want the coloration markup to be part of the translatable string, so that the unit name can have white text both before and after?
Yes, that's definitely needed if parts of the string are formatted differently. Can only be ignored if the entire string is formatted in the same way. So for `<i>I am an unit</i>` I don't need the markup, but for `I am an <i>unit</i>`, I need it, because 'unit' might be placed somewhere else in the translated version.

**Comment by babaissarkar** [comment]
@CelticMinstrel That's also fine. If the translation can work without having to bother about formatting, that's even better, but it might require more reorganizing in the codebase.

**Comment by babaissarkar** [comment]
2f4d6bafe2098e56996151e8034156a6691cdb6e also contains a couple of small bugfixes.
@stevecotton is it fine to mark the alignment strings as untranslatable? they are basically values that are inserted directly into the WML. Look at this :
Screenshot from 2024-04-16 20-14-41

Another weird issue :
Screenshot from 2024-04-16 20-15-21

Translated text gets inserted into the WML when using a different language.

**Comment by soliton-** [comment]
That's a bug. WML code should not be translated.

**Comment by babaissarkar** [comment]
Bugfixes moved to #8846 including https://github.com/wesnoth/wesnoth/pull/8731#issuecomment-2059282680.

**Comment by Pentarctagon** [comment]
Is there anything remaining for this PR aside from resolving the conflicts?

**Comment by babaissarkar** [comment]
> Is there anything remaining for this PR aside from resolving the conflicts?

Probably not.
@stevecotton Are the changes in this PR still relevant? I think most of the translation teams have already translated the new strings in Unit Type Editor, in which case this becomes unnecessary.

**Comment by stevecotton** [comment]
> Are the changes in this PR still relevant? I think most of the translation teams have already translated the new strings in Unit Type Editor, in which case this becomes unnecessary.

I'm confused - AFAICS, the Unit Type Editor is only in 1.19. The translation teams generally only work on 1.18.

**Comment by babaissarkar** [comment]
@stevecotton can this be merged, or you'd suggest waiting some more?

File: src/gui/dialogs/editor/edit_unit.cpp
Comment (stevecotton) [inline src/gui/dialogs/editor/edit_unit.cpp:69]: This isn't the place for a Doxygen comment, it should either stay as `//` or be `/* .... */`.

**Comment by stevecotton** [comment]
Thinking about 1.20, for non-editable values, I think this should use `heading: $value`, with coloration included in `$value`, both ideas as said above.

But, when the translatable string is a label next to an editbox, I guess it needs a version that doesn't have the `$value`, and shouldn't have a trailing space (and just assume that it's in the editor, so not worry about bidirectional text for this). My feeling is to ask @irydacea whether to put such strings in `wesnoth` or `wesnoth-lib`, but to get a complete set in one domain.

File: src/gui/dialogs/editor/edit_unit.hpp
Comment (soliton-) [inline src/gui/dialogs/editor/edit_unit.hpp:62]: Why can't you use `unit_alignments::values`?

Comment (babaissarkar) [reply src/gui/dialogs/editor/edit_unit.hpp]: Didn't know about that, updated.

File: src/gui/dialogs/editor/edit_unit.cpp
Comment (babaissarkar) [reply src/gui/dialogs/editor/edit_unit.cpp]: Updated.

**Comment by Wedge009** [comment]
Where do we stand here? Looks like it's pretty close to merging - if not already there - but I don't know if there are still outstanding concerns.

**Comment by babaissarkar** [comment]
> Where do we stand here? Looks like it's pretty close to merging - if not already there - but I don't know if there are still outstanding concerns.

Waiting for @stevecotton's approval.

**Comment by Vultraz** [comment]
Looking at this again, it doesn’t seem great that some labels use colons and others do not. Personally, I’d say don’t merge.

**Comment by babaissarkar** [comment]
> Looking at this again, it doesn’t seem great that some labels use colons and others do not. Personally, I’d say don’t merge.

I think I agree that inconsistent colons do lead to bad UI design, and we probably need a couple more strings with colons. What about the non-colon changes, or if I make the colons consistent? (though we might need some more string)

File: src/gui/dialogs/editor/edit_unit.cpp
Comment (stevecotton) [inline src/gui/dialogs/editor/edit_unit.cpp:92]: This could be a range-based for loop.

**Comment by stevecotton** [comment]
Clarifying a previous comment of mine, we shouldn't have trailing spaces in `"Property:"` strings, even if that cleanup means that we have near-duplicates that differ only by the trailing space for a while.

I'm still not sure whether they should be in `wesnoth` or `wesnoth-lib`, but we should have the complete set of strings for showing the properties of a unit, all in one text domain or the other.

File: src/gui/dialogs/editor/edit_unit.cpp
Comment (babaissarkar) [reply src/gui/dialogs/editor/edit_unit.cpp]: Updated.

File: data/gui/themes/default/window/editor_edit_unit.cfg
Comment (stevecotton) [inline data/gui/themes/default/window/editor_edit_unit.cfg:258]: This line seems to be a leftover

File: data/gui/themes/default/window/editor_edit_unit.cfg
Comment (stevecotton) [inline data/gui/themes/default/window/editor_edit_unit.cfg:222]: I'm against having trailing spaces in strings like this. Even if that currently means that they can't reuse a string, the eventual aim of the cleanup should be to remove the trailing space from the existing translatable strings.

Comment (babaissarkar) [reply data/gui/themes/default/window/editor_edit_unit.cfg:222]: Sure

**Comment by babaissarkar** [comment]
@stevecotton I can move this entirely into the `wesnoth` textdomain and remove the trailing spaces.

**Comment by babaissarkar** [comment]
> At least for the moment, my feeling is to take the commit that uses the existing unit_alignment data, but drop the commit with the string changes.

I've dropped that commit (and replaced another occurrence of `vgettext_impl`)

File: src/gui/dialogs/editor/edit_unit.cpp
Comment (stevecotton) [inline src/gui/dialogs/editor/edit_unit.cpp:111]: This (and the existing implementation) only get into the .po files because the same string is in theme.cpp, but it's not a regression so let's not worry about it for now.

**Review by stevecotton — APPROVED** [review]

File: data/gui/themes/default/window/editor_edit_unit.cfg
Comment (babaissarkar) [reply data/gui/themes/default/window/editor_edit_unit.cfg]: corresponding commit dropped.

**Pull request merged**
